### ISSUE TITLE
Language switch button

### ISSUE THREAD

**Issue by jo-spek** [google/ground-android#2826]
Ground gets displayed in English / French / Spanish / Portuguese based on the users system locale. 

If it is not too much work to implement at the moment, it would be nice to have a language switch button.

(First suggested in https://github.com/google/ground-android/issues/2676. Other FR closed because it was also about French translations which exist now.)

**Comment by gino-m** [comment]
This would mostly likely be an option in the settings screen rather than a button per se. Flagging for review.

**Comment by jo-spek** [comment]
> This would mostly likely be an option in the settings screen, and perhaps a dropdown during sign-in rather than a button per se. Flagging for review.

Oh yes, this doesn't have to be a button specifically. Just a setting that can be triggered one way or another.

**Comment by gino-m** [comment]
> > This would mostly likely be an option in the settings screen, and perhaps a dropdown during sign-in rather than a button per se. Flagging for review.
> 
> Oh yes, this doesn't have to be a button specifically. Just a setting that can be triggered one way or another.

Iiuc @kenstershiro will be responsible for turning these feature requests into actual CUJs, but might be helpful to start thinking in those terms anyway: https://carlinyuen.medium.com/framing-ux-concepts-for-your-team-cujs-use-cases-personas-e5767a3878a6

tl;dr: <who> <wants to achieve what goal> <optionally: any why>, for ex: Data collectors what to choose a language besides their device language because <explain why device language might not be the language they want>.

**Comment by kenstershiro** [comment]
Hey @jo-spek , could you help provide use cases where this would be useful, in order to prioritise the feature request appropriately? A couple I can think of are: 
a) As a tester of Ground app and website 
I want to be able to easily change language from the site/app 
so that I can test the coverage and correctness of terms in another language
b) As a user with a device locale in a different language to that being used by other survey participants
I want to be able to easily change language from the site/app 
so that I can see the survey in the same language as participants

**Comment by kenstershiro** [comment]
Thanks @jo-spek , icebox or keep it as a P2? I think another advantage of a language selector is making it clearer which languages are supported. By defaulting to the system language users don't know the possibility exists to change language.

**Comment by gino-m** [comment]
@jo-spek Gentle ping - should this be in the Icebox, or implemented before an eventual GA release?

**Comment by jo-spek** [comment]
I think this should be P1. Just have a simple language selector in the settings. Not hyper-important right now, since it's based on system locale, but there are many circumstances where users should have the possibility to switch.

**Comment by kenstershiro** [comment]
Question: Language selection takes place after sign in. If user sets language and then sings out, question on whether we conserve the language set or default back to phone lang.
Decision: Conserve the language set by the user, even after sign out.

**Comment by gino-m** [comment]
@kenstershiro @jo-spek  Should we also allow prompt the user for the language when signing in? In that case keeping the previous selection might make more sense, since it wouldn't be hidden / mysterious. Wdyt?

**Comment by jo-spek** [comment]
@gino-m @kenstershiro Yeah, why not. Sounds like a good idea to me. Having one of these well-known language button graphics on the start screen would be neat. 

Such as the one used by Wikipedia: ![Image](https://github.com/user-attachments/assets/f6761979-8bed-413f-8a29-e9c671be2c62)

**Comment by kenstershiro** [comment]
Hy @gino-m and @jo-spek . I was looking for an app on my phone that does this, but I couldn't find an app with a language selector on sign in. Suggestion, given this is already quite an edge case, we could start with a language selector in Settings, and if someone raises the need for it consider adding one on sign in as a follow up story?

**Comment by gino-m** [comment]
@kenstershiro Good idea!

**Comment by jo-spek** [comment]
@kenstershiro Sure, sounds good.

**Comment by kenstershiro** [comment]
Follow up is #3138 for the selector on sign in page
